src/mlpack/core/optimizers/smorms3/smorms3_update.hpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file smorms3_update.hpp
+ * @author Vivek Pal
+ *
+ * SMORMS3 update for Stochastic Gradient Descent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SMORMS3_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SMORMS3_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * SMORMS3 is an optimizer that estimates a safe and optimal distance based on
+ * curvature and normalizing the stepsize in the parameter space. It is a hybrid
+ * of RMSprop and Yann LeCun’s method in "No more pesky learning rates".
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @misc{Funk2015,
+ *   author    = {Simon Funk},
+ *   title     = {RMSprop loses to SMORMS3 - Beware the Epsilon!},
+ *   year      = {2015}
+ * }
+ * @endcode
+ */
+
+class SMORMS3Update
+{
+ public:
+  /**
+   * Construct the SMORMS3 update policy with given epsilon parameter.
+   *
+   * @param epsilon Value used to initialise the mean squared gradient parameter.
+   */
+  SMORMS3Update(const double epsilon = 1e-16) :
+              epsilon(epsilon),
+              previousStepSize(0)
```

Comment:
The spacing here seems weird, the initialization lists should only be 2 indents in if they are on newlines.